MQ2FeedMe.cpp

```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
 {
 	char Buffer[MAX_STRING*10];
 	MyList->clear();
-	if(GetPrivateProfileString(fSec,NULL,"",Buffer,MAX_STRING*10,INIFileName)) {
+
+	if(GetPrivateProfileString(fSec, NULL, "", Buffer, MAX_STRING * 10 , INIFileName))
+	{
 		char  szTemp[MAX_STRING];
-		PCHAR pBuffer=Buffer;
-		while (pBuffer[0]!=0) {
-			GetPrivateProfileString(fSec,pBuffer,"",szTemp,MAX_STRING,INIFileName);
-			if(szTemp[0]!=0) MyList->push_back(std::string(szTemp));
-			pBuffer+=strlen(pBuffer)+1;
+		char* pBuffer = Buffer;
+
+		while (pBuffer[0] !=0 )
+
+		{
+			GetPrivateProfileString(fSec, pBuffer, "", szTemp, MAX_STRING, INIFileName);
+			if (szTemp[0] != 0)
+			{
+				MyList->push_back(std::string(szTemp));
+			}
+
+			pBuffer += strlen(pBuffer) + 1;
```

Comment:
This section seems ugly and archaic. Pretty sure we have much better methods of doing this now.

Reply:
We do. `GetPrivateProfileKeys` would replace all of this I believe.